File: Retrofit2/app/src/main/java/com/example/retrofit2/MainActivity.java
package com.example.retrofit2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    String baseUrl = "https://jsonplaceholder.typicode.com/";

    TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv = findViewById(R.id.tv);

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        Api api = retrofit.create(Api.class);

        Call<List<DataModel>> call = api.getDataModels();

        call.enqueue(new Callback<List<DataModel>>() {
            @Override
            public void onResponse(Call<List<DataModel>> call, Response<List<DataModel>> response) {
                List<DataModel > data = response.body();

                for(int i=0;i<data.size();i++)
                {
                    tv.append("Id :"+data.get(i).getId()+
                    "\n Title :"+data.get(i).getTitle()+
                    "\n Body :"+data.get(i).getBody()+"\n\n\n\n");
                }
            }

            @Override
            public void onFailure(Call<List<DataModel>> call, Throwable t)
            {
                Toast.makeText(getApplicationContext(), t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });

    }
}

File: RetrofitRecyclerViewK/app/src/main/java/com/example/retrofitrecyclerviewk/MainActivity.kt
package com.example.retrofitrecyclerviewk

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import android.widget.Toast
import org.w3c.dom.Text
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.lang.StringBuilder

const val BASE_URL = "https://jsonplaceholder.typicode.com/"
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        getMyData()
    }

    private fun getMyData() {

        val retrofitBuilder = Retrofit.Builder()
            .addConverterFactory(GsonConverterFactory.create())
            .baseUrl(BASE_URL)
            .build()
            .create(ApiInterface::class.java)

        val retrofitData = retrofitBuilder.getData()

        retrofitData.enqueue(object : Callback<List<DataModels>?> {
            override fun onResponse(
                call: Call<List<DataModels>?>,
                response: Response<List<DataModels>?>
            ) {
                val responseBody = response.body()!!

                val myStringBuilder = StringBuilder()

                for(myData in responseBody)
                {
                    myStringBuilder.append(myData.id)
                    myStringBuilder.append("\n")

                }
                val textId = findViewById<TextView>(R.id.textId)
                textId.text = myStringBuilder

            }

            override fun onFailure(call: Call<List<DataModels>?>, t: Throwable) {
               Log.d("Error",""+t.message)
            }
        })
    }
}